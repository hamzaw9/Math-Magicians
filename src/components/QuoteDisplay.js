import { useEffect, useState } from 'react';

const QuoteDisplay = () => {
  const [quote, setQuote] = useState('');
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState('');
  useEffect(() => {
    (async () => {
      try {
        const data = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=happiness',
          {
            method: 'GET',
            headers: {
              'X-Api-Key': 'nDzwqTka6swjBQQbFR+ujA==lQfhCoGOjI3pUI5v',
            },
            contentType: 'application/json',
          },
        );
        const result = await data.json();
        setQuote(result[0].quote);
      } catch (error) {
        setError(error);
      }
    })();
  }, []);
  useEffect(() => {
    setIsLoading(false);
  }, [quote]);
  return (
    <div className="quote-container">
      <p className="quote">{isLoading ? 'Loading' : quote}</p>
      <p className="error">{error}</p>
    </div>
  );
};

export default QuoteDisplay;
