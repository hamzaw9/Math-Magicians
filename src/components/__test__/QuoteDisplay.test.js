import { render } from '@testing-library/react';
import QuoteDisplay from '../QuoteDisplay';

describe('QuoteDisplay', () => {
  it('QuoteDisplay component matches snapshot', () => {
    const { asFragment } = render(<QuoteDisplay />);
    expect(asFragment()).toMatchSnapshot();
  });
});
