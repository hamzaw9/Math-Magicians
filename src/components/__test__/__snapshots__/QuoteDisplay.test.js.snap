// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteDisplay QuoteDisplay component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="quote-container"
  >
    <p
      class="quote"
    />
    <p
      class="error"
    />
  </div>
</DocumentFragment>
`;
